### ISSUE TITLE
Destructor called too early when chaining get_database and get_layer

### ISSUE THREAD

**Issue by kb173** [boku-ilen/geodot-plugin#59]
Sometimes it would be convenient to chain Geodot calls like this:

```
var layer = Geodot.get_dataset(path).get_layer(name)
```

But this produces an error: The resulting layer does not point to a valid OGRLayer, so the program crashes. A workaround is to do this:

```
var dataset = Geodot.get_dataset(path)
var layer = dataset.get_layer(name)
```

It seems like when chaining, some destructor is called too early. I assume the dataset gets deleted, so the layer doesn't point to anything valid anymore.

**Comment by kb173** [comment]
A hotfix was pushed in https://github.com/boku-ilen/geodot-plugin/commit/910242285d6a2b6424d561e245bbfb28fa29e83b, but these destructor calls need to be re-implemented properly!

**Comment by kb173** [comment]
Fixed by smart pointers in e44a349 and proper destructor re-implementations in bdd82f1
